engine/src/main/java/org/pentaho/di/trans/steps/webservices/wsdl/Wsdl.java

```diff
@@ -0,0 +1,401 @@
+   */
+  public Wsdl( URI wsdlURI, QName serviceQName, String portName ) throws AuthenticationException {
+    this( wsdlURI, serviceQName, portName, null, null );
+  }
+
+  public Wsdl( URI wsdlURI, QName serviceQName, String portName, String username, String password ) throws AuthenticationException {
+
+    this.wsdlURI = wsdlURI;
+    try {
+      _wsdlDefinition = parse( wsdlURI, username, password );
+    } catch ( AuthenticationException ae ) {
+      // throw this again since KettleException is catching it
+      throw ae;
+    } catch ( WSDLException e ) {
+      throw new RuntimeException( "Could not load WSDL file: " + e.getMessage(), e );
+    } catch ( KettleException e ) {
+      throw new RuntimeException( "Could not load WSDL file: " + e.getMessage(), e );
+    }
+    if ( serviceQName == null ) {
+      _service = (Service) _wsdlDefinition.getServices().values().iterator().next();
+    } else {
+      _service = _wsdlDefinition.getService( serviceQName );
+      if ( _service == null ) {
+        throw new IllegalArgumentException( "Service: "
+          + serviceQName + " is not defined in the WSDL file " + wsdlURI );
+      }
+    }
+
+    if ( portName == null ) {
+      _port = getSoapPort( _service.getPorts().values() );
+    } else {
+      _port = _service.getPort( portName );
+      if ( _port == null ) {
+        throw new IllegalArgumentException( "Port: "
+          + portName + " is not defined in the service: " + serviceQName );
+      } else {
+        _port = _service.getPort( portName );
+      }
+    }
+
+    _wsdlTypes = new WsdlTypes( _wsdlDefinition );
+    _operationCache = new HashMap<String, WsdlOperation>();
+  }
+
+  /**
+   * Returns the first Soap port from the passed collection of Ports.
+   *
+   * @param portCollection
+   * @return
+   */
+  private Port getSoapPort( Collection<?> portCollection ) {
+    Port soapPort = null;
+    Iterator<?> iterator = portCollection.iterator();
+    while ( iterator.hasNext() ) {
+      Port port = (Port) iterator.next();
+      if ( WsdlUtils.isSoapPort( port ) ) {
+        soapPort = port;
+        break;
+      }
+    }
+    return soapPort;
+  }
+
+  /**
+   * Loads and parses the specified WSDL file.
+   *
+   * @param wsdlLocator
+   *          A javax.wsdl.WSDLLocator instance.
+   * @param serviceQName
+   *          Name of the service in the WSDL.
+   * @param portName
+   *          The service port name.
+   */
+  public Wsdl( WSDLLocator wsdlLocator, QName serviceQName, String portName ) throws AuthenticationException {
+    this( wsdlLocator, serviceQName, portName, null, null );
+  }
+
+  public Wsdl( WSDLLocator wsdlLocator, QName serviceQName, String portName, String username, String password ) throws AuthenticationException {
+
+    // load and parse the WSDL
+    try {
+      _wsdlDefinition = parse( wsdlLocator, username, password );
+    } catch ( AuthenticationException ae ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)